core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -714,4 +714,16 @@ public static String generateBadRecordsPath(String badLogStoreLocation, String s
           CarbonCommonConstants.FILE_SEPARATOR + taskNo;
     }
   }
+
+  /**
+   * Return the parent path of the input file
+   */
+  public static String getParentPath(String dataFilePath) {
+    int endIndex = dataFilePath.lastIndexOf(CarbonCommonConstants.FILE_SEPARATOR);
```

Comment:
CarbonCommonConstants.FILE_SEPARATOR is the last character of dataFilePath, such as '/user/warehouse/carbon.store/www/'

Reply:
the input is a file path like `/user/warehouse/carbon.store/www/file.carbondata`, so this function will return `/user/warehouse/carbon.store/www/`

Reply:
I updated the comment in this function